python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -830,6 +836,55 @@ def addDimensionEntry(self, dimension, dataId=None, entry=None, **kwds):
             self.setDimensionRegion(dataId)
         return dataId
 
+    @disableWhenLimited
+    @transactional
+    def addDimensionEntryList(self, dimension, dataIdList, entry=None, **kwds):
+        # Docstring inherited from Registry.addDimensionEntryList.
+        dataIdList = [DataId(dataId, dimension=dimension, universe=self.dimensions, **kwds) for dataId in
+                      dataIdList]
+        # The given dimension should be the only leaf dimension of the graph,
+        # and this should ensure it's a true `Dimension`, not a `str` name.
+        # all of dataIds should share a common dimension, so use the first
+        dimension, = dataIdList[0].dimensions().leaves
+        if entry is not None:
```

Comment:
Does there need to be a check that all entries in `dataIdList` have the same set of `dimensions`?  The case I'm worried about is based on the DM-19271 skymap change: if all `tract` and `patch` entries are contained in a single list (due to a think-o or something similar), does this fail noticeably?

Reply:
That would fail noticeably